Documentation/WorkflowSwiftUI Adoption Guide.md

```diff
@@ -0,0 +1,448 @@
+
+        func apply(toState state: inout State) -> Never? {
+            state.counter -= 1
+            return nil
+        }
+    }
+
+    func makeInitialState() -> State {
+        .init()
+    }
+
+    func render(
+        state: State,
+        context: RenderContext<Self>
+    ) -> CustomModel {
+        CustomModel(
+            accessor: context.makeStateAccessor(state: state),
+            child: TrivialWorkflow().rendered(in: context),
+            up: context.makeSink(of: UpAction.self),
+            down: context.makeSink(of: DownAction.self)
+        )
+    }
+}
+
+struct CustomModel: ObservableModel {
+    var accessor: StateAccessor<ComplexWorkflow.State>
+
+    var child: TrivialWorkflow.Rendering
+
+    var up: Sink<ComplexWorkflow.UpAction>
+    var down: Sink<ComplexWorkflow.DownAction>
+}
+```
+
+## Views
+
+In your View, you’ll access state and sinks via a property of type `Store<Model>`. This type wraps the model that your workflow renders, and provides conveniences for access the state, sinks, and child stores.
+
+If targeting iOS 16 or below, you’ll need to wrap your view’s body in `WithPerceptionTracking`. This is a component of the `Perception` backport that allows observation to work on iOS before iOS 17.
+
+```swift
+struct PersonState {
+  var name: String
+  var age: Int
+}
+
+enum PersonAction {
+  case giveHighFive
+}
+
+typealias PersonModel = ActionModel<PersonState, PersonAction>
+
+struct PersonView: View {
+  let store: Store<PersonModel>
+  
+  var body: some View {
+    WithPerceptionTracking {
+      VStack {
+        Text("Name: \(store.name)")
+        Text("Age: \(store.age)")
+
+        Button("✋") {
+          store.send(.giveHighFive)
+        }
+      }
+    }
+  }
+}
+```
+
+If you forget to wrap your view's body in `WithPerceptionTracking`, a runtime warning will remind you.
+
+## Screens
+
+In `WorkflowSwiftUI`, we recommend that workflows render models rather than screens. This strategy will help you to build smaller workflows and views that are more composable and reusable.
+
+When you are ready to put a view in a screen, create a screen that implements `ObservableScreen`.
+
+```swift
+public protocol ObservableScreen: Screen {
+    /// The type of the root view rendered by this screen.
+    associatedtype Content: View
+    /// The type of the model that this screen observes.
+    associatedtype Model: ObservableModel
+
+    /// The model that this screen observes.
+    var model: Model { get }
+
+    /// Constructs the root view for this screen. This is only called once to initialize the view.
+    /// After the initial construction, the view will be updated by injecting new values into the
+    /// store.
+    @ViewBuilder
+    static func makeView(store: Store<Model>) -> Content
+}
+```
+
+Your screen will be relatively simple, as it has the relatively simple job of acting as the glue between your rendering model and your view type.
+
+```swift
+struct PersonScreen: ObservableScreen {
```

Comment:
I think I mentioned this, but it'd be neat if we had a generic type that folks could use instead of needing to define this on their own